rand_core/src/block.rs

```diff
@@ -123,7 +123,7 @@ pub trait CryptoBlockRng: BlockRngCore {}
         bound = "for<'x> R: Serialize + Deserialize<'x> + Sized, for<'x> R::Results: Serialize + Deserialize<'x>"
     )
 )]
-pub struct BlockRng<R: BlockRngCore + ?Sized> {
+pub struct BlockRng<R: BlockRngCore> {
```

Comment:
This is changed because of the following warning:
```
warning: `?Sized` bound is ignored because of a `Sized` requirement
note: ...because `Deserialize` has the bound `Sized`
```
AFAIK we can not use this trait with DSTs either way, since we store core by value, so the `?Sized` bound is not needed.

Reply:
Rust does permit the last field of a struct to be unsized, but I can't see it being useful.

In this case we can drop the `+ Sized` bound three lines up.

Reply:
>Rust does permit the last field of a struct to be unsized,

Huh, TIL. I guess it's not surprising I did not encounter it in practice since it's a hardly useful feature in its current form and even docs state that:

>Yes, custom DSTs are a largely half-baked feature for now.